### ISSUE TITLE
Heterogeneous Graph Edge Classification, Predicting an Edge Type and Testing

### ISSUE THREAD

**Issue by NANGIA70** [dmlc/dgl#4404]
Hello, I would like to ask you about predicting the edge type of an existing edge with heterogeneous graphs.
I am referring to https://docs.dgl.ai/en/0.8.x/guide/training-edge.html the article provided in this link.
I am a beginner and unable to figure out how you are getting the final prediction from this(2 or 3). 
Moreover I am unable to figure out how to do validation and testing on this.

Thank you very much for your reply!

**Comment by BarclayII** [comment]
> I am a beginner and unable to figure out how you are getting the final prediction from this(2 or 3).

What are you referring to by saying 2 or 3?

> Moreover I am unable to figure out how to do validation and testing on this.

There shouldn't be that different between training and validation/test.  Essentially in the training loop you replace the loss computation with metric computation (such as accuracy), and `dec_graph` with only the validation/test edges.

**Comment by peizhou001** [comment]
Hi, do you have any further questions? We will close it, feel free to reopen it if any further questions.
